src/V3AstNodes.cpp

```diff
@@ -277,10 +277,10 @@ int AstBasicDType::widthTotalBytes() const {
 
 bool AstBasicDType::same(const AstNode* samep) const {
     const AstBasicDType* const sp = static_cast<const AstBasicDType*>(samep);
-    if (!rangep() && !sp->rangep() && m == sp->m) {
-        return true;
+    if (!rangep() && !sp->rangep()) {
+        return m == sp->m && numeric() == sp->numeric();
     } else {
-        return m == sp->m && rangep() && rangep()->sameTree(sp->rangep());
+        return rangep() && rangep()->sameTree(sp->rangep());
```

Comment:
Why removing the `m` compare when rangep() is non-null?

Reply:
Sorry it's a mistake... Now fixed